python/dgl/dataloading/negative_sampler.py

```diff
@@ -65,3 +66,67 @@ def _generate(self, g, eids, canonical_etype):
         src = F.repeat(src, self.k, 0)
         dst = F.randint(shape, dtype, ctx, 0, g.number_of_nodes(vtype))
         return src, dst
+
+# Alias
+Uniform = PerSourceUniform
+
+class GlobalUniform(_BaseNegativeSampler):
+    """Negative sampler that randomly chooses negative source-destination pairs according
+    to a uniform distribution.
+
+    For each edge ``(u, v)`` of type ``(srctype, etype, dsttype)``, DGL generates
+    :attr:`k` pairs of negative edges ``(u', v')``, where ``u'`` is chosen uniformly from
+    all the nodes of type ``srctype`` and ``v'`` is chosen uniformly from all the nodes
+    of type ``dsttype``.  The resulting edges will also have type
+    ``(srctype, etype, dsttype)``.
+
+    Parameters
+    ----------
+    k : int
+        The number of negative examples to generate per minibatch.
```

Comment:
"examples" -> "samples"

Reply:
We should perhaps rename this to `k_max`, or something similar to indicate that `k` samples are not guaranteed.

Reply:
I'm hesitant to name it `k_max` because most of the time it does generate `k` examples, barring some corner cases where the graph is extremely dense.  Naming it to `k_max` may give people impression that it often does not generate `k` examples.